ui/components/app/confirm/info/row/url.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import {
+  Box,
+  Icon,
+  IconName,
+  IconSize,
+  Text,
+} from '../../../../component-library';
+import {
+  AlignItems,
+  BorderRadius,
+  Display,
+  FlexWrap,
+  IconColor,
+  TextColor,
+  TextVariant,
+  BackgroundColor,
+} from '../../../../../helpers/constants/design-system';
+
+export type ConfirmInfoRowUrlProps = {
+  url: string;
+};
+
+export const ConfirmInfoRowUrl = ({
+  url: urlString,
+}: ConfirmInfoRowUrlProps) => {
+  let url;
+
+  try {
+    url = new URL(urlString);
+  } catch (e) {
+    console.log(`ConfirmInfoRowUrl: Error parsing url: ${urlString}`);
+  }
+
+  const isHTTP = url?.protocol === 'http:';
+  const isValidUrl = Boolean(url);
+
+  const displayUrl = isValidUrl
+    ? `${url?.host}${url?.pathname === '/' ? '' : url?.pathname}`
+    : urlString;
```

Comment:
I think we should not parse URL as above to get display string, we should display the url string as it is.

Reply:
sounds good to me. updated [df5ba34](https://github.com/MetaMask/metamask-extension/pull/22293/commits/df5ba34cb1e2671754107d4dd9a3e2ec126056bc)